godot-core/src/builtin/meta/signature.rs

```diff
@@ -74,40 +76,31 @@ macro_rules! impl_varcall_signature_for_tuple {
             where $R: VariantMetadata + ToVariant + sys::GodotFuncMarshal + Debug,
                $( $Pn: VariantMetadata + FromVariant + sys::GodotFuncMarshal + Debug, )*
         {
-            #[inline]
-            fn variant_type(index: i32) -> sys::VariantType {
+            const ARGUMENTS_LEN: usize = $ARGUMENTS_LEN;
+
+            fn argument_info(index: usize, param_name: &str) -> Option<MethodArgumentInfo> {
                 match index {
-                    -1 => $R::variant_type(),
                     $(
-                        $n => $Pn::variant_type(),
+                        $n => Some($Pn::argument_info(param_name)),
                     )*
-                    _ => unreachable!("variant_type: unavailable for index {}", index),
-                    //_ => sys::GDEXTENSION_VARIANT_TYPE_NIL
+                    _ => None,
                 }
             }
```

Comment:
The only place where this is used immediately `unwrap()`s the option, so you get now a worse error message than before.

I would panic inside here instead of returning an `Option`. Probably `panic!` is better than `unreachable!` though.

Reply:
Well if the `unwrap()` fails then there's a bug somewhere anyway. Since `argument_info` should always return `Some(_)` as long as the `index` is in the range `0..PARAM_COUNT`.

And i feel like it makes more sense to panic when we actually know for sure that `index` is within the range `0..PARAM_COUNT`.

This would also make it easier to safely perform the optimization of `unwrap_unchecked` if we ever decide to do that.

I can definitely add a better error message when i perform the unwrap though.